core/src/main/java/org/bitcoinj/base/BitcoinNetwork.java

```diff
@@ -84,21 +86,33 @@ public Coin maxMoney() {
      * @throws IllegalArgumentException if there is no matching enum
      */
     public static BitcoinNetwork of(String nameString) {
-        return find(nameString)
+        return findCanonicalName(nameString)
                 .orElseThrow(() -> new IllegalArgumentException("Unrecognized network name : " + nameString));
     }
 
     /**
-     * Find the {@code BitcoinNetwork} from a name String
+     * Find the {@code BitcoinNetwork} from a name string, or from a number of common alternate names
      * @param nameString A name string (e.g. "MAIN", "TEST", "SIGNET")
      * @return An {@code Optional} containing the matching enum or empty
      */
     public static Optional<BitcoinNetwork> find(String nameString) {
+        final Optional<BitcoinNetwork> canonicalName = findCanonicalName(nameString);
+        if (canonicalName.isPresent()) return canonicalName;
+        return findAlternateName(nameString);
+    }
+
+    private static Optional<BitcoinNetwork> findCanonicalName(String nameString) {
         return Arrays.stream(values())
                 .filter(n -> n.toString().equals(nameString.toUpperCase()))
```

Comment:
I'm having second thoughts about the `.toUpperCase()` here and in `findAlternateName`. We are not providing an option for someone to find an exact, case-sensitive match. There are three alternatives:
1. Require the caller to use `.toUpperCase`, e.g. `var optionalNetwork = find(nameCandidate.toUpperCase())`
2. Provide an overloaded `find` method with a 2nd, boolean parameter called `ignoreCase`, eg. `var optionalNetwork = find(nameCandidate, true)`
3. Leave it as is and see if we really need it.

Reply:
I strongly dislke 2. Imho "canonical" should not allow for variance.

For similar reason, I dislike 3 as well because if we need it then we're faced with incompatible code.

I think we should make sure the `of()` methods require the exect name/ID. To me, the question is do we want all-uppercase or all-lowercase names (IDs are already all-lowercase).